plugins/inputs/tencentcloudcm/tencentcloudcm.go

```diff
@@ -0,0 +1,448 @@
+package tencentcloudcm
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/limiter"
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common"
+	tcerrors "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common/errors"
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common/profile"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// TencentCloudCM contains the configuration for the Tencent Cloud Cloud Monitor plugin.
+type TencentCloudCM struct {
+	Endpoint string `toml:"endpoint"`
+
+	Period config.Duration `toml:"period"`
+	Delay  config.Duration `toml:"delay"`
+
+	RateLimit int             `toml:"ratelimit"`
+	Timeout   config.Duration `toml:"timeout"`
+
+	DiscoveryInterval config.Duration `toml:"discovery_interval"`
+
+	Accounts []*Account `toml:"accounts"`
+
+	client cmClient
+
+	Log telegraf.Logger `toml:"-"`
+
+	windowStart  time.Time
+	windowEnd    time.Time
+	discoverTool *DiscoverTool
+}
+
+// Account defines a Tencent Cloud account
+type Account struct {
+	Name       string       `toml:"name"`
+	SecretID   string       `toml:"secret_id"`
+	SecretKey  string       `toml:"secret_key"`
+	Namespaces []*Namespace `toml:"namespaces"`
+
+	Crs *common.Credential
+}
+
+// Namespace defines a Tencent Cloud CM namespace
+type Namespace struct {
+	Name    string    `toml:"namespace"`
+	Metrics []string  `toml:"metrics"`
+	Regions []*Region `toml:"regions"`
+}
+
+// Region defines a Tencent Cloud region
+type Region struct {
+	RegionName string      `toml:"region"`
+	Instances  []*Instance `toml:"instances"`
+}
+
+// Instance defines a generic Tencent Cloud instance
+type Instance struct {
+	Dimensions []*Dimension `toml:"dimensions"`
+}
+
+// Dimension defines a simplified Tencent Cloud Cloud Monitor dimension.
+type Dimension struct {
+	Name  string `toml:"name"`
+	Value string `toml:"value"`
+}
```

Comment:
Why do we need to define `Dimension`? Can't we simply use `mnap[string][string]`?
```suggestion
type Instance struct {
	Dimensions map[string]string `toml:"dimensions"`
}
```